src/module.c

```diff
@@ -10336,11 +10336,17 @@ int RM_Scan(RedisModuleCtx *ctx, RedisModuleScanCursor *cursor, RedisModuleScanC
     }
     int ret = 1;
     ScanCBData data = { ctx, privdata, fn };
-    cursor->cursor = dictScan(ctx->client->db->dict, cursor->cursor, moduleScanCallback, &data);
+    int slot = getAndClearSlotIdFromCursor(&cursor->cursor);
+    dict *dict = ctx->client->db->dict[slot];
+    cursor->cursor = dictScan(dict, cursor->cursor, moduleScanCallback, &data);
     if (cursor->cursor == 0) {
-        cursor->done = 1;
-        ret = 0;
+        dict = dbGetNextUnvisitedSlot(ctx->client->db, &slot);
+        if (dict == NULL) {
+            cursor->done = 1;
+            ret = 0;
+        }
     }
+    addSlotIdToCursor(slot, &cursor->cursor);
```

Comment:
To be honest, I don't think this would add much value or convenience as there is not much logic to hide there. Let me know if you feel strongly about this.